app/components/LargeHeader.js

```diff
@@ -19,7 +26,7 @@ function LargeHeader() {
         <div className="search-container form-row" role="search">
           <input type="text" className="search-field" placeholder="I'm looking for..." name="srch-term" id="srch-term" />
           <input type="text" className="location-field" placeholder="Near me" name="srch-location" id="srch-location" />
-          <button id="largeheader_searchbutton" className="button search" type="submit"><i className="material-icons">search</i> <span>Search</span></button>
+          <button onClick={submitSearch} id="largeheader_searchbutton" className="button search" type="submit"><i className="material-icons">search</i> <span>Search</span></button>
```

Comment:
I don't have time to test this, but to get functions working I usually need something like onClick={this.myFunction.bind(this)}. Also if you use submit you might need to capture the event and call preventDefault for example submitSearch(e){ e.preventDefault(); and then more functionality...}